internal/chart/pvc.go

```diff
@@ -0,0 +1,63 @@
+package chart
+
+import (
+	"context"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	dockerRegistryPVCName = "serverless-docker-registry"
+	pvcKind               = "PersistentVolumeClaim"
+	pvcVersion            = "v1"
+	pvcGroup              = ""
+)
+
+func AdjustDockerRegToClusterPVCSize(ctx context.Context, c client.Client, obj unstructured.Unstructured) (unstructured.Unstructured, error) {
+	if obj.GetName() != dockerRegistryPVCName {
+		return obj, nil
+	}
+	clusterPVC := corev1.PersistentVolumeClaim{}
+	err := c.Get(ctx, client.ObjectKey{
+		Namespace: obj.GetNamespace(),
+		Name:      obj.GetName(),
+	}, &clusterPVC)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			return obj, nil
+		}
+		return obj, errors.Wrap(err, "while getting pvc from cluster")
+	}
+	objPVC := corev1.PersistentVolumeClaim{}
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(obj.Object, &objPVC); err != nil {
+		return obj, errors.Wrap(err, "while converting unstructured to pvc")
+	}
+	storage := clusterPVC.Spec.Resources.Requests.Storage()
+	if storage.Equal(*objPVC.Spec.Resources.Requests.Storage()) {
+		return obj, nil
+	}
+	objPVCcopy := objPVC.DeepCopy()
+	objPVCcopy.Spec.Resources.Requests[corev1.ResourceStorage] = *clusterPVC.Spec.Resources.Requests.Storage()
+
+	out, err := runtime.DefaultUnstructuredConverter.ToUnstructured(objPVCcopy)
+	if err != nil {
+		return obj, errors.Wrap(err, "while converting copied pvc object to unstructured")
+	}
```

Comment:
you could direclty change the value on the unstruct w/o deep copying and conversion.
see
```
err = unstructured.SetNestedField(obj.Object, storage.String(), "spec", "resources", "requests", "storage")
```

Reply:
Yes, but I also need to handle error.
When I want to get value using `NestedField`I also need to handle error, found bool. 
It won't make less code, but it also make the code less bullet proof to changes in core api.